packages/credential-store/src/types/filters.ts

```diff
@@ -7,16 +8,24 @@ import { CredentialRole, DigitalCredential, DocumentType, FindDigitalCredentialA
  * @param idOrHash - The ID or hash of the credential to search for.
  * @returns A FindDigitalCredentialArgs array for filtering by ID or hash.
  */
-export const credentialIdOrHashFilter = (credentialRole: CredentialRole, idOrHash: string): FindDigitalCredentialArgs => [
-  {
-    id: idOrHash,
-    credentialRole,
-  },
-  {
-    hash: idOrHash,
-    credentialRole,
-  },
-]
+
+export const credentialIdOrHashFilter = (credentialRole: CredentialRole, idOrHash: string): FindDigitalCredentialArgs => {
+  const filter: FindDigitalCredentialArgs = [
+    {
+      hash: idOrHash,
+      credentialRole,
+    },
+  ]
+
+  if (uuidValidate(idOrHash)) {
+    filter.push({
+      id: idOrHash,
```

Comment:
Credential id's certainly do not have to be UUIDs

Reply:
Then it is searching the wrong id. The record id is of type uuid
```
    await queryRunner.query(`
      CREATE TABLE "DigitalCredential" (
        "id" uuid NOT NULL DEFAULT uuid_generate_v4(),
```

Do we need a separate field credentialId in the DB? The VC's id is in the document string which can only be searched after parsing the JSON.

Reply:
Or the method name is wrong and we do want to search on the PK

Reply:
I think both. Being able to search for the id of the credential itself (not pk) makes sense to me,  just like the id (pk) in the db. Probably in the old store they were one and the same

Reply:
I'll have to hoist the credentialId out of the credential to be able to search on it.

Reply:
added credentialId: idOrHash above (next commit)